File: CSC205-Java/_solutions/Module05-Recursion-and-Analysis-of-Algorithms/debug01/Factorial.java
// Computes n! recursively for several values of n.
public class Factorial {
    public static long factorial(int n) {
        if (n <= 1) {
            return 1;
        }
        return n * factorial(n - 1);
    }

    public static void main(String[] args) {
        int[] inputs = {5, 3, 1, 0};
        for (int n : inputs) {
            System.out.println(n + "! = " + factorial(n));
        }
    }
}

File: CSC205-Cpp/Module05-Recursion-and-Analysis-of-Algorithms/debug01/factorial.cpp
// factorial.cpp
// Recursive factorial with a small driver.

#include <iostream>
using namespace std;

// Returns n! for n >= 0.
long long factorial(int n) {
    if (n == 1) {
        return 1;
    }
    return n * factorial(n - 1);
}

int main() {
    int inputs[] = { 5, 10, 1, 0 };

    for (int n : inputs) {
        cout << n << "! = " << factorial(n) << endl;
    }
    return 0;
}
